modules/apps/commerce/commerce-product-asset-categories-navigation-web/src/main/resources/META-INF/resources/configuration.jsp

```diff
@@ -129,13 +129,17 @@ if (assetVocabulary != null) {
 
 			var form = window.document['<portlet:namespace />fm'];
 
-			var assetCategoryIdsKey = Object.keys(form.elements).filter((input) =>
-				input.includes('assetCategoryIds')
+			var assetCategoryIdsKey = Object.keys(form.elements).filter(
+				(input) => input.includes('assetCategoriesSelectorCategoryId')
 			);
 
-			preferencesRootAssetCategoryId.val(
-				form.elements[assetCategoryIdsKey].value
-			);
+			for (const assetCategoryId of assetCategoryIdsKey) {
+				if (form.elements[assetCategoryId].value) {
+					preferencesRootAssetCategoryId.val(
+						form.elements[assetCategoryId].value);
+					break;
+				}
+			}
```

Comment:
We do not want to use `for...of` loops.

[Here's one file that's overriding the rule](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/utilities/forms/formsHelper.js#L28-L42), which we don't want to do.